lib/livebook_web/live/session_live.ex

```diff
@@ -614,6 +614,7 @@ defmodule LivebookWeb.SessionLive do
         id="export"
         session={@session}
         tab={@tab}
+        has_stale_cell?={@has_stale_cell?}
```

Comment:
https://github.com/livebook-dev/livebook/blob/6f4eb7f58bb8b55f7aff7562dab719a14a58aa29/lib/livebook_web/live/session_live/export_component.ex#L10-L18

In the export component we fetch the notebook (because the LV prunes cell sources and outputs from memory), and we do that only once. Consequently we should use `assign_new` for the stale flag, so that it doesn't get out of sync (currently it updates when changing tabs).

Sidenote: while at it, you can also change the `:notebook` if to `assign_new` :)

Reply:
like this?
https://github.com/livebook-dev/livebook/pull/2420/files#diff-a1ce1bab35b3b99f014dd7b48df188336f07cf5c87941ab8c3c4dd0e771f80fdR14

Reply:
Yes, we need to do the same with the stale flag :)